### DISCUSSION TITLE
Custom Fields (timesheet) in invoice

### DISCUSSION THREAD

**Discussion by derdoc800** [kimai/kimai#4714] [Questions & Answers] [answered]
I am trying to use custom fields (plugin) in invoice, ideally in docx but when following instruction as per documentation ${entry.meta.foo} it is not getting resolved. The custom field is visible.

When using the HTML sample (https://www.kimai.org/documentation/invoices.html#more-invoice-templates)
i get some entry.*.meta.* objects but not the ones attached to the timesheets entry.meta.foo 

Anybody has experienced this behavior. Should we log an issue?

**Comment by kevinpapst** [comment, marked as answer]
Your invoice template is not allowed to use grouping.

It must be configured to use:
<img width="328" alt="Bildschirmfoto 2024-03-22 um 09 05 39" src="https://github.com/kimai/kimai/assets/533162/c7b28197-2536-4fa7-aa51-27012906f9b8">

**Reply by derdoc800** [reply]
Thank you for your quick answer. .. understood ... That means we cannot use the approach we wanted to ... to group invoices per custom field. We will see if we get to the desired result some other way.

**Reply by kevinpapst** [reply]
Try to explain what you want to achieve and I'll see if there is a (simple) way.

**Reply by derdoc800** [reply]
Ok let me try to explain:

Even though we have only one defined activity with the customer e.g. Support he wants to have his invoices split on different departments for their internal crosscharging. Also our timesheets (pdf custom twig) are split based on this invoice_topic. Also our goal was to use only docx templates that our internal accounting can create and adjust those templates themselves.

What we did so far:
1.) Created a custom field invoice_topic for each time entry and limited security for that field to this specific customer
2.) We did not want to use tags as they cannot provide a limited drop down list.
3.) Invoice amounts can be filtered on the invoice_topic:HR

Now what would like to achieve is that this Invoice topic is displayed in the invoice right next to the order number or in a seperate entry column but of course we do not want all detailled entries in the invoice.

**Reply by kevinpapst** [reply]
First feedback:
2) You can by removing the permission `create_tag` for the normal user / specific user role at System > Roles
